### DISCUSSION TITLE
Packaging WebVtt subtitles on HLS LL CMAF2 from a MPEGTS stream

### DISCUSSION THREAD

**Discussion by estigma88** [OvenMediaLabs/OvenMediaEngine#1560] [Ideas]
**Use Case**

We have an MPEGTS stream with three tracks: video, audio, subtitles/data. We want to use OvenMedia packager to take that stream and generate HLS LL, with CMAF2 on video and audio tracks, and plain Webvtt as subtitles track.

We would like to know if OvenMedia supports for that:
1. Does OvenMedia support subtitles input?
2. Does OvenMedia support HLS LL packaging with CMAF2?
3. Does OvenMedia support MPEGTS input with multiple tracks?
4. Can we have MPEGTS with only audio and video tracks, and have a different input of the packager with the subtitles stream?

We explored the OvenMedia documentation and couldn't find a full response of those questions. We are open to contribute if those features are not implemented yet.

**Comment by getroot** [comment]
- Subtitles support in MPEG-2 TS provider and LL-HLS publisher is in the backlog but is a low priority. (Because there are not many use cases in live broadcasts)
- Supports LL-HLS with CMAF. Is CMAF2 a different specification? I have never heard of this yet. I would be grateful if you could provide me with a related link.
- OME can ingest multiple audio tracks over MPEG-2 TS. But not a subtitle.
- Currently OME does not parse subtitles from MPEG-2 TS. (What specific specifications does it come with?)
Using the Mux Provider you can merge multiple stream inputs into one. https://airensoft.gitbook.io/ovenmediaengine/live-source/multiplex-channel

**Comment by estigma88** [comment]
> backlog but is a low priority

Could you give me some pointers to the backlog if it is public? also, we might contribute to that work if you provide some initial guidance on where to start

 > Supports LL-HLS with CMAF. Is CMAF2 a different specification?

I read about it here https://wiki.gpac.io/Howtos/dash/cmaf/?h=cmaf, not sure if it is accurate

> Currently OME does not parse subtitles from MPEG-2 TS. (What specific specifications does it come with?)

My understanding is that MPEGTS supports dvb_teletext, but, as we might be able to pass any plain text/data as a track, I am hoping to have a data track with WebVTT directly there, or just using the multiplex channel to join the video and audio stream with a WebVTT stream
